core/src/main/java/org/frankframework/receivers/Receiver.java

```diff
@@ -1146,9 +1146,17 @@ public void retryMessage(String storageKey) throws ListenerException {
 				IMessageBrowser<?> errorStorageBrowser = messageBrowsers.get(ProcessState.ERROR);
 				IbisTransaction itx = new IbisTransaction(txManager, getTxDef(), "receiver [" + getName() + "]");
 				try {
-					RawMessageWrapper<?> msg = errorStorageBrowser.browseMessage(storageKey);
 					//noinspection unchecked
-					processRawMessage((RawMessageWrapper<M>) msg, session, true, false);
+					RawMessageWrapper<M> msg = (RawMessageWrapper<M>) errorStorageBrowser.browseMessage(storageKey);
+					if (listener instanceof IHasProcessState<?>) {
+						//noinspection unchecked
+						IHasProcessState<M> hasProcessState = (IHasProcessState<M>) listener;
+						if (hasProcessState.knownProcessStates().contains(ProcessState.INPROCESS)) {
+
+							msg = hasProcessState.changeProcessState(msg, ProcessState.INPROCESS, "Message manually retried");
+						}
+					}
```

Comment:
Moving message from "error" to "in process" state was fix for the original exception reported, because now it was again possible to move the file to "error" on subsequent error.

Reply:
I hope it's a bit clearer in the code now?